tests/test_sshfs.py

```diff
@@ -71,6 +72,19 @@ def strip_keys(info):
         info.pop(key, None)
 
 
+def test_fsspec_registration(fs, ssh_server, remote_dir, user="user"):
+    """Check that we can use sshfs with fsspec.open and an ssh url."""
+    path = remote_dir + "/a.txt"
+
+    # Additional options like client_keys need to be passed to open.
+    url = f"ssh://{user}@{ssh_server.host}:{ssh_server.port}/{path}"
+    with fsspec.open(url, "w", client_keys=[USERS[user]]) as file:
+        file.write("Anything")
+
+    with fs.open(path, "r") as file:
+        assert file.read() == "Anything"
```

Comment:
Can we be sure that this is sshfs and not sftpfs?

Reply:
Good point. I'll investigate if there is any way to assert that the correct file system is created.

Reply:
I found a way. You can access the underlying filesystem used by fsspec via `file.buffer.fs`. I added asserts for both the class and parameters.